corehq/apps/users/models.py

```diff
@@ -1678,6 +1678,16 @@ def unset_location(self):
         self.get_domain_membership(self.domain).location_id = None
         self.save()
 
+    def assign_to_locations(self, location_ids):
+        self.assigned_location_ids = location_ids
+        self.get_domain_membership(self.domain).assigned_location_ids = location_ids
+        self.save()
+
+    def unassign_all_locations(self):
+        self.assigned_location_ids = []
+        self.get_domain_membership(self.domain).assigned_location_ids = []
+        self.save()
+
```

Comment:
Yeah, I don't like this duplicate code either, but it's hard to reuse it without using some hacky function/attribute-name inspect tooling, as the two methods differ, both in the way they get the attribute `assigned_location_ids` and the attributes that need to be updated. Keeping them separate is more cleaner.